### PR TITLE
Delete new row when cancel is clicked

### PR DESCRIPTION
**Pull request by rawagner**
https://bugzilla.redhat.com/show_bug.cgi?id=1646953

### CODE REVIEW

File: src/components/Wizard/CreateVmWizard/CreateVmWizard.js [hunk 1]
```diff
@@ -116,9 +116,10 @@ export class CreateVmWizard extends React.Component {
 
   lastStepReached = () => this.state.activeStepIndex === this.getLastStepIndex();
 
-  onStepDataChanged = (tabKey, value, valid) => {
+  onStepDataChanged = (tabKey, value, valid, editing = false) => {
     const validatedTabData = validateTabData(tabKey, value, valid);
     this.safeSetState(state => ({
+      editing,
```
Comment (suomiy) [inline src/components/Wizard/CreateVmWizard/CreateVmWizard.js]: I think the tab should merely request to lock the createVmWizard and not take Wizard state into account. This is very useful when using asynchronous actions.

I suggest something like:

```js
{
  stepData: {
    [tabKey]: {
      lockStep
    }
  }
}

```

File: src/components/Table/EditableDraggableTable.js [hunk 2]
```diff
@@ -96,17 +97,24 @@ class EditableDraggableTable extends React.Component {
...
-      const index = findIndex(rows, { id });
-
-      rows[index] = cloneDeep(rows[index].backup);
-      delete rows[index].backup;
+      const index = findIndex(this.props.rows, { id });
+      let rows;
+
+      let type = ON_CANCEL;
+      if (this.props.rows[index].newRow) {
+        rows = this.props.rows.filter(row => row.id !== id);
```
Comment (suomiy) [inline src/components/Table/EditableDraggableTable.js]: can't we rather call onDelete method from here?

Comment (rawagner) [reply src/components/Table/EditableDraggableTable.js]: extracted onDelete to separate function which is now reused where needed

File: src/components/Wizard/CreateVmWizard/CreateVmWizard.js (hunk 1 above)
Comment (rawagner) [reply src/components/Wizard/CreateVmWizard/CreateVmWizard.js]: done, `lockStep` is under `[tabKey]` now

**Pull request merged**